server/chat-plugins/the-studio.ts

```diff
@@ -212,6 +212,29 @@ export class LastFMInterface {
 	}
 }
 
+export class SpotifyInterface {
```

Comment:
Why is this class necessary? Just make it a regex, and use `RegExp.exec` to pull out the song ID.

Reply:
`/show` can handle the song display, and the regex can just be a `const`.

Reply:
if you have, say, https://open.spotify.com/track/1SUJo2OxP9QfEegvoXxILL?si=-ybc9PMBTH2u9phSvWekXA.
All you'd need (which would negate the use of this class) would be to just use 
```ts
const spotifyRegex = /open\.spotify\.com\/track\/(.*)/;
const possibleTrack = spotifyRegex.exec(target)?.[0];
if (possibleTrack) {
	buf += `<spotify src="https://open.spotify.com/embed/track/${possibleTrack}"></spotify>`;
} else ....
```

The regex will pull `1SUJo2OxP9QfEegvoXxILL?si=-ybc9PMBTH2u9phSvWekXA`, for you, and as a result can handle both the "is this a spotify url?" and "what's the song id?"

Reply:
You got a point, lemme just change after checking.

Reply:
Bte that regex didn't pull out `5h5eqopAnR9BuTdPerTkcM?si=A7KPiDlGQvuJKJMiao0q0g`

Reply:
You should do `spotifyRegex.exec(target)?.[1];`, but yes, it works.
<img width="692" alt="image" src="https://user-images.githubusercontent.com/56906084/100910587-0558d500-3483-11eb-8a06-103e45765e5e.png">